EddiInaraService/InaraService.cs

```diff
@@ -0,0 +1,282 @@
+            {
+                Dictionary<string, object> data = new Dictionary<string, object>()
+                {
+                    { "InaraAPIEvent", indexedEvents.Find(e => e.eventCustomID == inaraResponse.eventCustomID) },
+                    { "InaraResponse", inaraResponse }
+                };
+                if (inaraResponse.eventStatus == 400)
+                {
+                    // 400 - Error (you probably did something wrong, there are properties missing, etc. The event was skipped or whole batch cancelled on failed authorization.)
+                    Logging.Error("Inara responded with: " + inaraResponse.eventStatusText, JsonConvert.SerializeObject(data));
+                }
+                else
+                {
+                    // 202 - Warning (everything is OK, but there may be multiple results for the input properties, etc.)
+                    // 204 - 'Soft' error (everything was formally OK, but there are no results for the properties set, etc.)
+                    // Inara may also return null as it undergoes a nightly manintenance cycle where the servers go offline temporarily.
+                    Logging.Warn("Inara responded with: " + inaraResponse.eventStatusText ?? "(No response)", JsonConvert.SerializeObject(data));
+
+                }
+                return false;
+            }
+        }
+
+        public async void SendQueuedAPIEventsAsync(bool inBeta = false)
+        {
+            List<InaraAPIEvent> queue = new List<InaraAPIEvent>();
+            while (Instance.queuedAPIEvents.TryDequeue(out InaraAPIEvent pendingEvent))
+            {
+                queue.Add(pendingEvent);
+            }
+            if (queue.Count > 0)
+            {
+                await Task.Run(() => Instance.SendEventBatch(ref queue, inBeta));
+                Instance.lastSync = queue.Max(e => e.eventTimeStamp);
+                InaraConfiguration inaraConfiguration = InaraConfiguration.FromFile();
+                inaraConfiguration.lastSync = Instance.lastSync;
+                inaraConfiguration.ToFile();
+            }
+        }
+
+        public void EnqueueAPIEvent(InaraAPIEvent inaraAPIEvent)
+        {
+            if (!(inaraAPIEvent is null) && lastSync < inaraAPIEvent.eventTimeStamp)
+            {
+                Instance.queuedAPIEvents.Enqueue(inaraAPIEvent);
+            }
+        }
+
+        public void DequeueAPIEventsOfType(string eventName)
+        {
+            List<InaraAPIEvent> queue = new List<InaraAPIEvent>();
+            while (Instance.queuedAPIEvents.TryDequeue(out InaraAPIEvent pendingEvent))
+            {
+                if (pendingEvent.eventName != eventName)
+                {
+                    queue.Add(pendingEvent);
+                }
+            }
+            foreach (InaraAPIEvent inaraAPIEvent in queue)
+            {
+                Instance.EnqueueAPIEvent(inaraAPIEvent);
+            }
+        }
+    }
+
+    internal class InaraSendJson
```

Comment:
For "plain old data" I would probably have gone with struct, but it doesn't really matter.

Reply:
I'm a lot more familiar with class than struct. Can you elaborate on why you'd have chosen struct?